util/argo/argo.go

```diff
@@ -145,22 +144,10 @@ func ValidateRepo(ctx context.Context, spec *argoappv1.ApplicationSpec, repoClie
 	repoRes, err := db.GetRepository(ctx, spec.Source.RepoURL)
 
 	if err != nil {
-		if errStatus, ok := status.FromError(err); ok && errStatus.Code() == codes.NotFound {
-			// The repo has not been added to Argo CD so we do not have credentials to access it.
-			// We support the mode where apps can be created from public repositories. Test the
-			// repo to make sure it is publicly accessible
-			err = git.TestRepo(spec.Source.RepoURL, "", "", "", false)
-			if err != nil {
-				conditions = append(conditions, argoappv1.ApplicationCondition{
-					Type:    argoappv1.ApplicationConditionInvalidSpecError,
-					Message: fmt.Sprintf("No credentials available for source repository and repository is not publicly accessible: %v", err),
-				})
-			} else {
-				repoAccessable = true
-			}
-		} else {
-			return nil, "", err
-		}
+		conditions = append(conditions, argoappv1.ApplicationCondition{
+			Type:    argoappv1.ApplicationConditionInvalidSpecError,
+			Message: fmt.Sprintf("repository not accessible: %v", err),
+		})
```

Comment:
The TestRepo() check shouldn't be removed. Users need to understand when they are specifying an invalid repo, or when credentials are bad. However, the consequence of doing every time as part of `ValidateRepo` is that we are making one more `git ls-remote` call any time someone uses the `ApplicationService` API to do CRUD operations on apps.

Reply:
Ops. Fixing.

Reply:
Fixed.